plugins/modules/mso_tenant_site.py

```diff
@@ -0,0 +1,394 @@
+                payload['azureAccount'][0]['cloudApplication'] = []
+                payload['azureAccount'][0]['cloudActiveDirectory'] = []
+
+            if azure_access_type == 'unmanaged':
+                if not azure_subscription_id:
+                    mso.fail_json(msg="azure_subscription_id is required when in unmanaged mode.")
+                if not azure_application_id:
+                    mso.fail_json(msg="azure_application_id is required when in unmanaged mode.")
+                if not secret_key:
+                    mso.fail_json(msg="secret_key is required when in unmanaged mode.")
+                if not azure_active_directory_id:
+                    mso.fail_json(msg="azure_active_directory_id is required when in unmanaged mode.")
+                if not azure_active_directory_name:
+                    mso.fail_json(msg="azure_active_directory_name is required when in unmanaged mode.")
+                if not azure_credential_name:
+                    mso.fail_json(msg="azure_credential_name is required when in unmanaged mode.")
+                azure_account.update(
+                    accessType='credentials',
+                )
+                cloudApplication = dict(
+                    cloudApplicationId=azure_application_id,
+                    cloudCredentialName=azure_credential_name,
+                    secretKey=secret_key,
+                    cloudActiveDirectoryId=azure_active_directory_id
+                )
+                cloudActiveDirectory = dict(
+                    cloudActiveDirectoryId=azure_active_directory_id,
+                    cloudActiveDirectoryName=azure_active_directory_name
+                )
+                payload['azureAccount'][0]['cloudApplication'] = [cloudApplication]
+                payload['azureAccount'][0]['cloudActiveDirectory'] = [cloudActiveDirectory]
+
+        else:
+            aws_account = dict(
+                accountId=cloud_account,
+                isTrusted=aws_trusted,
+                accessKeyId=aws_access_key,
+                secretKey=secret_key,
+                isAccountInOrg=aws_account_org,
+            )
+
+            if not aws_trusted:
+                if not aws_access_key:
+                    mso.fail_json(msg="aws_access_key is a required field in untrusted mode.")
+                if not secret_key:
+                    mso.fail_json(msg="secret_key is a required field in untrusted mode.")
+            payload['awsAccount'] = [aws_account]
+
+    sites = [(s.get('siteId')) for s in mso.query_objs('tenants')[tenant_idx]['siteAssociations']]
+
+    if site_id not in sites:
+        mso.existing = {}
+    else:
+        site_idx = sites.index((site_id))
+        mso.existing = mso.query_objs('tenants')[tenant_idx]['siteAssociations'][site_idx]
+
+    if state == 'query':
+        if len(sites) == 0:
+            mso.fail_json(msg="No site associated with tenant Id {0}".format(tenant_id))
+        elif site_id in sites:
```

Comment:
unnecessary condition

Reply:
If I remove this condition and query a tenant with no sites associated with it, user encounters the following error:
 **None is not in list\n**

It would be neater to have a check and display the appropriate error.

Reply:
What do you suggest?